the-skycoin-project/skywire
# [WIP] Skychat Refactoring

With this commit the app now has got a structure like proposed from [Panayiotis Kritiotis](https://pkritiotis.io/clean-architecture-in-golang/)
It was rewritten in a way to enable group chats in the future.

Also at the moment repositories are only implemented in memory so user and chat data will be lost when stopping the app.

There are still bugs to fix like unmarshalling and changing behavior when interchanging information between the two chat members.

#### Did you run `make format && make check`?
no

### Fixes #	

### Changes:	
The basic connecting works but there are still problems with sending and receiving user info to the chat member.

### How to test this PR:
Best to test this app and to see the received bytes is to use the current skychat app from v1.0.1 (which displayes the sent json directly) and two instances of this commit to test behaviours.
